### PR TITLE
fixed issue #522

### PR DESCRIPTION
**Pull request by user-promi**

### CODE REVIEW

File: api/class-mvx-rest-controller.php [hunk 1]
```diff
@@ -2640,6 +2640,57 @@ public function mvx_bulk_todo_pending_product($request) {
                     }
                 }
             }
+        } elseif ($type == "pending_verification") {
+            if ($this->mvx_fetch_pending_verification_data() && !empty($this->mvx_fetch_pending_verification_data()->data)) {
+                foreach ($this->mvx_fetch_pending_verification_data()->data as $key => $value) {
+                    if (!empty($data_list[$key])) {
+                        $get_pending_verification_list[] = $value['id'];
```
Comment (abhirup97) [inline api/class-mvx-rest-controller.php:2647]: why you checked $data_list[$key] you using the $value['id'] variable??

File: api/class-mvx-rest-controller.php [hunk 2]
```diff
@@ -2640,6 +2640,57 @@ public function mvx_bulk_todo_pending_product($request) {
...
+                    }
+                }
+            }
+            if ($select_option_value == "dismiss") {
```
Comment (abhirup97) [inline api/class-mvx-rest-controller.php:2651]: $select_option_value make sure this variable has data just check your self.

**Review by abhirup97 — CHANGES_REQUESTED** [review]

File: api/class-mvx-rest-controller.php [hunk 3]
```diff
@@ -2640,6 +2640,57 @@ public function mvx_bulk_todo_pending_product($request) {
...
+                        }
+                        update_user_meta($user_id, 'mvx_vendor_verification_settings', wc_clean($verification_settings));
+                        if (isset($verification_settings['address_verification']['is_verified']) && $verification_settings['address_verification']['is_verified'] == 'verified' && isset($verification_settings['id_verification']['is_verified']) && $verification_settings['id_verification']['is_verified'] == 'verified') {
+                            update_user_meta($user_id, 'mvx_vendor_is_verified', 'verified');
+                        } else {
+                            delete_user_meta($user_id, 'mvx_vendor_is_verified');
+                        }
+                    }
+                }
+            } else {
```
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php:2674]: why else only? next line if?
why?

File: api/class-mvx-rest-controller.php [hunk 4]
```diff
@@ -2640,6 +2640,57 @@ public function mvx_bulk_todo_pending_product($request) {
...
+                        $verification_settings = get_user_meta($user_id, 'mvx_vendor_verification_settings', true);
+                        if (isset($verification_settings['address_verification']['is_verified'])) {
+                            $verification_settings['address_verification']['is_verified'] = 'verified';
+                        }
+                        if (isset($verification_settings['id_verification']['is_verified'])) {
+                            $verification_settings['id_verification']['is_verified'] = 'verified';
+                        }
+                        update_user_meta($user_id, 'mvx_vendor_verification_settings', wc_clean($verification_settings));
+                        if (isset($verification_settings['address_verification']['is_verified']) && $verification_settings['address_verification']['is_verified'] == 'verified' && isset($verification_settings['id_verification']['is_verified']) && $verification_settings['id_verification']['is_verified'] == 'verified') {
+                            update_user_meta($user_id, 'mvx_vendor_is_verified', 'verified');
```
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php:2687]: update without sanitize?

File: api/class-mvx-rest-controller.php [hunk 5]
```diff
@@ -2640,6 +2640,57 @@ public function mvx_bulk_todo_pending_product($request) {
...
+                            delete_user_meta($user_id, 'mvx_vendor_is_verified');
+                        }
+                    }
+                }
+            } else {
+                if ($get_pending_verification_list) {
+                    foreach ($get_pending_verification_list as $request_vendor_key => $request_vendor_id) {
+                        $user_id = $request_vendor_id;
+                        $verification_settings = get_user_meta($user_id, 'mvx_vendor_verification_settings', true);
+                        if (isset($verification_settings['address_verification']['is_verified'])) {
```
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php:2679]: if $verification_settings return a blank array then?

File: api/class-mvx-rest-controller.php [hunk 6]
```diff
@@ -2640,6 +2640,57 @@ public function mvx_bulk_todo_pending_product($request) {
...
+                            $verification_settings['address_verification']['is_verified'] = 'verified';
+                        }
+                        if (isset($verification_settings['id_verification']['is_verified'])) {
+                            $verification_settings['id_verification']['is_verified'] = 'verified';
```
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php:2683]: if $verification_settings return a blank array then?

File: api/class-mvx-rest-controller.php [hunk 7]
```diff
@@ -2456,7 +2456,7 @@ public function mvx_bulk_todo_pending_product($request) {
         $data_list = $request && $request->get_param('data_list') ? $request->get_param('data_list') : '';
         $select_option_value = $request && $request->get_param('value') ? $request->get_param('value') : '';
         $type = $request && $request->get_param('type') ? $request->get_param('type') : '';
-        $get_product_list = $get_users_list = $get_coupon_list = $get_transaction_list = $get_pending_questions_list = [];
+        $get_product_list = $get_users_list = $get_coupon_list = $get_transaction_list = $get_pending_questions_list = $get_pending_verification_list = [];
         if ($type == "pending_product") {
```
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php:2460]: where u have check for $type

File: api/class-mvx-rest-controller.php [hunk 8]
```diff
@@ -2640,6 +2640,57 @@ public function mvx_bulk_todo_pending_product($request) {
                     }
                 }
             }
+        } elseif ($type == "pending_verification") {
```
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php:2643]: where u have check for $type

File: api/class-mvx-rest-controller.php (hunk 2 above)
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php:2651]: where u have check for $select_option_value

File: api/class-mvx-rest-controller.php [hunk 9]
```diff
@@ -2640,6 +2640,57 @@ public function mvx_bulk_todo_pending_product($request) {
...
+                        $get_pending_verification_list[] = $value['id'];
+                    }
+                }
+            }
+            if ($select_option_value == "dismiss") {
+                if ($get_pending_verification_list) {
+                    foreach ($get_pending_verification_list as $request_vendor_key => $request_vendor_id) {
+                        $user_id = $request_vendor_id;
+                        $verification_settings = get_user_meta($user_id, 'mvx_vendor_verification_settings', true);
+                        if (isset($verification_settings['id_verification']['is_verified']) && $verification_settings['id_verification']['is_verified'] != 'verified') {
```
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php:2656]: where u have check for $verification_settings

File: api/class-mvx-rest-controller.php [hunk 10]
```diff
@@ -2640,6 +2640,57 @@ public function mvx_bulk_todo_pending_product($request) {
...
+                            $verification_settings['id_verification']['is_verified'] = 'rejected';
+                            $verification_settings['id_verification']['data'] = '';
+                            delete_user_meta($user_id, 'mvx_vendor_is_verified');
+                        }
+                        if (isset($verification_settings['address_verification']['is_verified']) &&  $verification_settings['address_verification']['is_verified'] != 'verified') {
+                            $verification_settings['address_verification']['is_verified'] = 'rejected';
+                            $verification_settings['address_verification']['data'] = '';
+                            delete_user_meta($user_id, 'mvx_vendor_is_verified');
+                        }
+                        update_user_meta($user_id, 'mvx_vendor_verification_settings', wc_clean($verification_settings));
```
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php:2666]: why  wc_clean? why not other sanitize funtions. when wc_clean used?

File: api/class-mvx-rest-controller.php [hunk 11]
```diff
@@ -2640,6 +2640,57 @@ public function mvx_bulk_todo_pending_product($request) {
...
+                        if (isset($verification_settings['address_verification']['is_verified']) && $verification_settings['address_verification']['is_verified'] == 'verified' && isset($verification_settings['id_verification']['is_verified']) && $verification_settings['id_verification']['is_verified'] == 'verified') {
```
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php:2667]: checking?

File: api/class-mvx-rest-controller.php [hunk 12]
```diff
@@ -2640,6 +2640,57 @@ public function mvx_bulk_todo_pending_product($request) {
...
+                            update_user_meta($user_id, 'mvx_vendor_is_verified', 'verified');
```
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php:2668]: before update checking?

File: api/class-mvx-rest-controller.php [hunk 13]
```diff
@@ -2640,6 +2640,57 @@ public function mvx_bulk_todo_pending_product($request) {
...
+                        } else {
+                            delete_user_meta($user_id, 'mvx_vendor_is_verified');
```
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php:2670]: add checking before delete ..metadata_exists or not

File: api/class-mvx-rest-controller.php [hunk 14]
```diff
@@ -2640,6 +2640,57 @@ public function mvx_bulk_todo_pending_product($request) {
                     }
                 }
             }
+        } elseif ($type == "pending_verification") {
+            if ($this->mvx_fetch_pending_verification_data() && !empty($this->mvx_fetch_pending_verification_data()->data)) {
+                foreach ($this->mvx_fetch_pending_verification_data()->data as $key => $value) {
+                    if (!empty($data_list[$key])) {
```
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php:2646]: if $data_list not retunn array then? where is the checking?

File: api/class-mvx-rest-controller.php (hunk 1 above)
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php:2647]: why you are using if block. isn't possible ternary operator?

File: api/class-mvx-rest-controller.php [hunk 15]
```diff
@@ -2640,6 +2640,57 @@ public function mvx_bulk_todo_pending_product($request) {
...
+            }
+            if ($select_option_value == "dismiss") {
+                if ($get_pending_verification_list) {
+                    foreach ($get_pending_verification_list as $request_vendor_key => $request_vendor_id) {
+                        $user_id = $request_vendor_id;
+                        $verification_settings = get_user_meta($user_id, 'mvx_vendor_verification_settings', true);
+                        if (isset($verification_settings['id_verification']['is_verified']) && $verification_settings['id_verification']['is_verified'] != 'verified') {
+                            $verification_settings['id_verification']['is_verified'] = 'rejected';
+                            $verification_settings['id_verification']['data'] = '';
+                            delete_user_meta($user_id, 'mvx_vendor_is_verified');
```
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php:2659]: why delete without meta exist checking

File: api/class-mvx-rest-controller.php [hunk 16]
```diff
@@ -2640,6 +2640,57 @@ public function mvx_bulk_todo_pending_product($request) {
...
+                        }
+                        if (isset($verification_settings['address_verification']['is_verified']) &&  $verification_settings['address_verification']['is_verified'] != 'verified') {
```
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php:2661]: $verification_settings checking?
if it return null then?

File: api/class-mvx-rest-controller.php [hunk 17]
```diff
@@ -2640,6 +2640,57 @@ public function mvx_bulk_todo_pending_product($request) {
...
+                            $verification_settings['address_verification']['is_verified'] = 'rejected';
```
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php:2662]: sanitizeing?

File: api/class-mvx-rest-controller.php [hunk 18]
```diff
@@ -2640,6 +2640,57 @@ public function mvx_bulk_todo_pending_product($request) {
...
+                            $verification_settings['address_verification']['data'] = '';
+                            delete_user_meta($user_id, 'mvx_vendor_is_verified');
```
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php:2664]: checking before delete?

File: api/class-mvx-rest-controller.php (hunk 6 above)
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php:2683]: without sanitation?

File: api/class-mvx-rest-controller.php [hunk 19]
```diff
@@ -2640,6 +2640,57 @@ public function mvx_bulk_todo_pending_product($request) {
...
+                }
+            } else {
+                if ($get_pending_verification_list) {
+                    foreach ($get_pending_verification_list as $request_vendor_key => $request_vendor_id) {
+                        $user_id = $request_vendor_id;
+                        $verification_settings = get_user_meta($user_id, 'mvx_vendor_verification_settings', true);
+                        if (isset($verification_settings['address_verification']['is_verified'])) {
+                            $verification_settings['address_verification']['is_verified'] = 'verified';
+                        }
+                        if (isset($verification_settings['id_verification']['is_verified'])) {
```
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php:2682]: $verification_settings checking?
if blank array or not an array?

File: api/class-mvx-rest-controller.php [hunk 20]
```diff
@@ -2640,6 +2640,57 @@ public function mvx_bulk_todo_pending_product($request) {
...
+                            $verification_settings['id_verification']['is_verified'] = 'verified';
+                        }
+                        update_user_meta($user_id, 'mvx_vendor_verification_settings', wc_clean($verification_settings));
+                        if (isset($verification_settings['address_verification']['is_verified']) && $verification_settings['address_verification']['is_verified'] == 'verified' && isset($verification_settings['id_verification']['is_verified']) && $verification_settings['id_verification']['is_verified'] == 'verified') {
```
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php:2686]: $verification_settings?

File: api/class-mvx-rest-controller.php [hunk 21]
```diff
@@ -2640,6 +2640,57 @@ public function mvx_bulk_todo_pending_product($request) {
...
+                            update_user_meta($user_id, 'mvx_vendor_is_verified', 'verified');
+                        } else {
+                            delete_user_meta($user_id, 'mvx_vendor_is_verified');
```
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php:2689]: checking?

**Review by moumitahalder — CHANGES_REQUESTED** [review]

**Comment by moumitahalder** [comment]
lot of issue @user-promi

**Pull request closed without merging**